File: Code Generation/responses/cpp/gemini/easy/unique-3-digit-even-numbers.cpp
#include <vector>
#include <set>
#include <algorithm>

using namespace std;

class Solution {
public:
    int totalNumbers(vector<int> digits) {
        int n = digits.size();
        set<int> uniqueEvenNumbers;

        // Since the maximum size of digits is 10, O(N^3) is very efficient.
        // We use nested loops to pick three distinct indices i, j, k.
        for (int i = 0; i < n; ++i) {
            // First digit (hundreds place) cannot be zero.
            if (digits[i] == 0) continue;

            for (int j = 0; j < n; ++j) {
                if (j == i) continue;

                for (int k = 0; k < n; ++k) {
                    if (k == i || k == j) continue;

                    // Last digit (units place) must be even.
                    if (digits[k] % 2 == 0) {
                        int number = digits[i] * 100 + digits[j] * 10 + digits[k];
                        uniqueEvenNumbers.insert(number);
                    }
                }
            }
        }

        return uniqueEvenNumbers.size();
    }
};

File: Code Generation/responses/python/gemini/easy/unique-3-digit-even-numbers.py
from typing import List
import itertools

class Solution:
    def totalNumbers(self, digits: List[int]) -> int:
        unique_evens = set()
        for p in itertools.permutations(digits, 3):
            if p[0] != 0 and p[2] % 2 == 0:
                num = p[0] * 100 + p[1] * 10 + p[2]
                unique_evens.add(num)
        return len(unique_evens)